template/conftest.py

```diff
@@ -1,156 +1,15 @@
-                    fw_list.append(fw)
-                    ids.append(fw.device + " " + fw.version)
-
-        metafunc.parametrize("firmware", fw_list, ids=ids, scope="session")
-
-
-def prepare_speculos_args(firmware: Firmware, display: bool):
-    speculos_args = []
-
-    if display:
-        speculos_args += ["--display", "qt"]
-
-    app_path = app_path_from_app_name(APPS_DIRECTORY, APP_NAME, firmware.device)
-
-    return ([app_path], {"args": speculos_args})
-
-
-# Depending on the "--backend" option value, a different backend is
-# instantiated, and the tests will either run on Speculos or on a physical
-# device depending on the backend
-def create_backend(backend_name: str, firmware: Firmware, display: bool,
-                   log_apdu_file: Optional[Path]):
-    if backend_name.lower() == "ledgercomm":
-        return LedgerCommBackend(firmware=firmware, interface="hid", log_apdu_file=log_apdu_file)
-    elif backend_name.lower() == "ledgerwallet":
-        return LedgerWalletBackend(firmware=firmware, log_apdu_file=log_apdu_file)
-    elif backend_name.lower() == "speculos":
-        args, kwargs = prepare_speculos_args(firmware, display)
-        return SpeculosBackend(*args, firmware=firmware, log_apdu_file=log_apdu_file, **kwargs)
-    else:
-        raise ValueError(f"Backend '{backend_name}' is unknown. Valid backends are: {BACKENDS}")
-
-
-# This fixture will return the properly configured backend, to be used in tests.
-# As Speculos instantiation takes some time, this fixture scope is by default "session".
-# If your tests needs to be run on independent Speculos instances (in case they affect
-# settings for example), then you should change this fixture scope and choose between
-# function, class, module or session.
-@pytest.fixture(scope="session")
-def backend(backend_name, firmware, display, log_apdu_file):
-    with create_backend(backend_name, firmware, display, log_apdu_file) as b:
-        yield b
-
-
-@pytest.fixture
-def navigator(backend, firmware, golden_run):
-    if firmware.device.startswith("nano"):
-        return NanoNavigator(backend, firmware, golden_run)
-    elif firmware.device.startswith("stax"):
-        return StaxNavigator(backend, firmware, golden_run)
-    else:
-        raise ValueError(f"Device '{firmware.device}' is unsupported.")
-
-
-@pytest.fixture
-def use_only_on_backend(request, backend_name):
-    if request.node.get_closest_marker('use_on_backend'):
-        current_backend = request.node.get_closest_marker('use_on_backend').args[0]
-        if current_backend != backend_name:
-            pytest.skip(f'skipped on this backend: "{current_backend}"')
-
-
-def pytest_configure(config):
-    config.addinivalue_line(
-        "markers",
-        "use_only_on_backend(backend): skip test if not on the specified backend",
-    )
+# Pull all features from the base ragger conftest using the overridden configuration
+pytest_plugins = ("ragger.conftest.base_conftest", )
```

Comment:
can we add an easy link to use for user wanting to look at the conftest content without having to find the github repo, and then scroll in it? something like https://github.com/LedgerHQ/ragger/blob/develop/src/ragger/conftest.base_conftest.py

Reply:
Mmh not a bad idea but I would prefer linking to a documentation than a source file.. How about merging this version and we improve after ?